File: samples/java/TicTacToe/Server/src/main/java/systems/zlink/samples/tictactoe/server/play/infrastructure/zlink/spots/tictactoegamespot/handlers/TicTacToeGameCreatedHandler.java
package systems.zlink.samples.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.handlers;

import systems.zlink.framework.messaging.ZLinkMessage;
import systems.zlink.framework.spots.ZLinkSpotCreateResponse;
import systems.zlink.samples.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.TicTacToeGame;
import systems.zlink.samples.tictactoe.shared.contracts.TicTacToeGameCreateReq;

public final class TicTacToeGameCreatedHandler {
    public ZLinkSpotCreateResponse handle(TicTacToeGame game, ZLinkMessage request) {
        game.markCreated(request.decode(TicTacToeGameCreateReq.class));
        return ZLinkSpotCreateResponse.accept();
    }
}

File: samples/kotlin/TicTacToe/Server/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/server/play/infrastructure/zlink/spots/tictactoegamespot/handlers/TicTacToeGameCreatedHandler.kt
package systems.zlink.samples.kotlin.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.handlers

import systems.zlink.framework.messaging.ZLinkMessage
import systems.zlink.samples.kotlin.tictactoe.server.play.infrastructure.zlink.spots.tictactoegamespot.TicTacToeGame
import systems.zlink.samples.kotlin.tictactoe.shared.contracts.TicTacToeGameCreateReq

class TicTacToeGameCreatedHandler {
    fun handle(game: TicTacToeGame, request: ZLinkMessage) {
        game.markCreated(request.decode(TicTacToeGameCreateReq::class.java))
    }
}
